WordPress/gutenberg
# Confirmdialog: add custom button text

## Description
Follow up to https://github.com/WordPress/gutenberg/pull/37602#discussion_r793911945, this PR adds the ability to set custom label text for the confirm and cancel buttons in the experimental `ConfirmDialog` component.

cc @fullofcaffeine @ciampo 

## Testing Instructions
- Launch Storybook locally
- locate the ConfirmDialog component and select the new 'With Custom Button Labels" Story
- manipulate the new "cancel button" and "confirm button" knobs to test different text strings
- make sure all components tests pass, as this PR also adds a new test to confirm the specified button text is being rendered

## Checklist:
